consensus/polybft/contracts_initializer.go

```diff
@@ -38,7 +38,7 @@ func initStakeManager(polyBFTConfig PolyBFTConfig, transition *state.Transition)
 			BlsKey: key.ToBigInt(),
 		}
 
-		approveFn := &contractsapi.ApproveNativeERC20Fn{
+		approveFn := &contractsapi.ApproveNativeERC20MintableFn{
```

Comment:
Follow-up on @dusannosovic-ethernal's PR:

On this line, we should approve on `stakeTokenAddr` (should be retrieved from `PolyBFTConfig`), instead of native token:

```go
err = callContract(validator.Address, contracts.NativeERC20TokenContract, input, "NativeERC20.approve", transition)
```

This wasn't caught by tests, because we don't have one where `stakeToken != nativeToken`.

Reply:
Let's address this through a separate PR.